_drafts/sharing-gdb-scripts.md

```diff
@@ -0,0 +1,163 @@
+- No stale Wiki pages or README's. No questions in Slack about how to launch GDB now that pull-request #1234 has been merged.
+
+These are the ways I like to load GDB scripts from favorite to least favorite.
+
+#### Shared Shell Script
+
+You can commit a shell script that everyone on the team uses to launch GDB and debug the firmware.
+
+```bash
+#!/bin/bash
+
+SYMBOLS="$1"
+arm-none-eabi-gdb-py $SYMBOLS -ix=tools/gdb/memfault.gdbinit
+```
+
+Then a member of the team can run it with the following (assuming it's marked as executable):
+
+```
+$ ./gdb.sh build/output/symbols.elf
+```
+
+If Bash scripts aren't your jam (they aren't mine) and would prefer writing Python instead, you should check out my previous post on wrapping [GDB with Invoke]({% post_url 2019-08-27-building-a-cli-for-firmware-projects %}).
+
+### GDB in VSCode
+
+If you are using vanilla VSCode & GDB or using an extension with VSCode, you should likely be able to define an `autorun` section in your `launch.json` where you can place `source ...`.
+
+```json
+{
+  "version": "0.2.0",
+  "configurations": [
+    {
+      "type": "gdb",
+      "request": "launch",
+      "name": "Launch Program",
+      "target": "./build/${workspaceFolderBasename}.elf",
+      "cwd": "${workspaceFolder}",
+      "gdbpath": "/opt/esp/xtensa-esp32-elf/bin/xtensa-esp32-elf-gdb",
+      "autorun": [
+        "source path/to/project.gdbinit",
+        "other...commands"
+      ],
+      "preLaunchTask": "openocd"
+    }
+  ]
+}
+```
+
+> You can also use `"debugger_args": ["-ix", "source path/to/project.gdbinit"]`.
+
+If you are using [Cortex-Debug](https://marketplace.visualstudio.com/items?itemName=marus25.cortex-debug), you would add a `preLaunchCommands` section to your `launch.json`.
+
+```json
+{
+    "version": "0.2.0",
+    "configurations": [
+        {
+            "type": "cortex-debug",
+            "request": "launch",
+            "name": "Debug (OpenOCD)",
+            "servertype": "openocd",
+            "preLaunchCommands": [
+                "source path/to/project.gdbinit",
+            ]
+        }
+    ]
+}
+```
+
+For [PlatformIO](https://platformio.org/), you can add this hook to your `platform.ini` under the `[env:debug]` section:
+
+```ini
+platform = ...
+board = ...
+debug_extra_cmds =
+  source path/to/project.gdbinit
+```
+
+You can then commit the `launch.json` file to the repo and all team members can load it.
+
+### GDB in Eclipse Embedded CDT
+
+You should be able to add the custom GDB hooks under *Debug Configuration* -> *GDB .* Debugging* -> *Your Configuration* -> *GDB Client Setup* -> *Commands*
+
+<p align="center">
+  <img width="600" src="{% img_url sharing-gdb-scripts/eclipse-debug-configuration.png %}" />
+</p>
+
+Here, you would insert something along the lines of:
+
+```
+source path/to/project.gdbinit
```

Comment:
I'm pretty sure one can even make use of eclipse predefined variables here: https://help.eclipse.org/2021-03/index.jsp?topic=%2Forg.eclipse.platform.doc.user%2Fconcepts%2Fconcepts-exttools.htm

`${project_loc}/project.gdbinit`